elixir/apps/api/lib/api/relay/socket.ex

```diff
@@ -11,34 +12,41 @@ defmodule API.Relay.Socket do
 
   @impl true
   def connect(%{"token" => encrypted_secret} = attrs, socket, connect_info) do
-    %{
-      user_agent: user_agent,
-      x_headers: x_headers,
-      peer_data: peer_data
-    } = connect_info
-
-    real_ip = API.Sockets.real_ip(x_headers, peer_data)
-
-    attrs =
-      attrs
-      |> Map.take(~w[ipv4 ipv6])
-      |> Map.put("last_seen_user_agent", user_agent)
-      |> Map.put("last_seen_remote_ip", real_ip)
-
-    with {:ok, token} <- Relays.authorize_relay(encrypted_secret),
-         {:ok, relay} <- Relays.upsert_relay(token, attrs) do
-      socket =
-        socket
-        |> assign(:relay, relay)
-
-      {:ok, socket}
-    else
-      {:error, :invalid_token} ->
-        {:error, :invalid_token}
-
-      {:error, reason} ->
-        Logger.debug("Error connecting relay websocket: #{inspect(reason)}")
-        {:error, reason}
+    :otel_propagator_text_map.extract(connect_info.trace_context_headers)
```

Comment:
@thomaseizinger @conectado FYI: if you decide to send W3C trace context headers - I will inherit the trace you started.

Reply:
I'll see to integrate that!